src/progress-bar/progress-bar.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import {render} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {create} from "react-test-renderer";
+import ProgressBar, {ProgressBarProps} from "./ProgressBar";
+import {testA11y} from "../core/utils/test/testUtils";
+
+describe("<ProgressBar />", () => {
+  const defaultProgressBarProps: ProgressBarProps = {
+    testid: "progress-bar",
+    percentage: 57,
+    style: {backgroundColor: "blue", completedColor: "yellow", trackColor: "orange"}
+  };
+
+  it("should render correctly", () => {
+    render(<ProgressBar {...defaultProgressBarProps} />);
+  });
+
+  it("should match snapshot", () => {
+    const tree = create(<ProgressBar {...defaultProgressBarProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    await testA11y(container);
+  });
+
+  it("should render children correctly", () => {
+    const progressBarContent = <span data-testid={"progress-bar.content"}>{"Test"}</span>;
+
+    const {getByTestId} = render(
+      <ProgressBar {...defaultProgressBarProps}>{progressBarContent}</ProgressBar>
+    );
+
+    expect(getByTestId(defaultProgressBarProps.testid!)).toContainElement(
+      getByTestId("progress-bar.content")
+    );
+  });
+
+  it("progress bar's background color should be backgroundColor", () => {
+    const {getByTestId} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    expect(getByTestId(defaultProgressBarProps.testid!)).toHaveStyle(
+      `background-color: ${defaultProgressBarProps.style.backgroundColor}`
+    );
+  });
+
+  it("the width of the progress bar track should be `percentage` percent", () => {
+    const {getByTestId} = render(<ProgressBar {...defaultProgressBarProps} />);
+
+    expect(getByTestId(`${defaultProgressBarProps.testid!}.track`)).toHaveStyle(
+      `width: ${defaultProgressBarProps.percentage}%`

```

Comment:
This seems to depend on the implementation. For example, the implementation details may change for the component but the visual output may still be the same. I wonder if we can decouple this from the implementation and test only the appearance of it without it relying on the implementation detail. May not be possible for this component but worth to ponder about.